packages/client/src/client4.ts

```diff
@@ -4262,6 +4264,13 @@ export default class Client4 {
             {method: 'get'},
         );
     }
+
+    deleteWorkspace = (deletionRequest: WorkspaceDeletionRequest) => {
+        return this.doFetch<StatusOK>(
+            `${this.getCloudRoute()}/delete-workspace`,
+            {method: 'post', body: JSON.stringify(deletionRequest)},
```

Comment:
Something to resolve between this PR and the backend PRs. This client currently does a POST with the body. The backend PRs listen for `DELETE`s and handle a body. `DELETE`'s [shouldn't have bodies in the request](https://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request), but some services do it anyways, including Mattermost api does it for 3 of 43 DELETE endpoints (`DeleteGroupMembers`, `RemoveTeamsFromRetentionPolicy`, `RemoveChannelsFromRetentionPolicy`), and CWS has a couple pre-existing DELETE endpoints that expect bodies as well.
 
I can think of 3 solutions:
* use POST method with data in body (webapp currently does this).
* use DELETE method with data in body (proposed backend implementations do this)
* use DELETE method but encode data as query string parameters.

My own two cents would be to use the DELETE method with data encoded as query string parameters, but I'd accept any of the 3 as long as both front and back end implementations agree on it.

For query string usage, we can do something like the cloud contact us link does: https://github.com/mattermost/mattermost-webapp/blob/master/selectors/cloud.ts#L47

Reply:
Going with delete, makes more sense. POST was just mindlessly adding this for testing.